### ISSUE TITLE
nvim_get_runtime_file is very slow when using a glob with `{A,B,C}`

### ISSUE THREAD

**Issue by ModProg** [neovim/neovim#18838]
### Neovim version (nvim -v)

v0.8.0-dev+207-g0adc66171

### Vim (not Nvim) behaves the same?

no, lua :)

### Operating system/version

Arch Linux

### Terminal name/version

wezterm 20220528-212940-84842aaa

### $TERM environment variable

xterm-256color

### Installation

AUR

### How to reproduce the issue

Compare how long
```sh
nvim -u NONE --cmd "lua vim.api.nvim_get_runtime_file('config/*.{yaml,yml,json}', true)" "+:q"
```
and
```sh
nvim -u NONE --cmd "lua vim.api.nvim_get_runtime_file('config/*.yaml', true) | lua vim.api.nvim_get_runtime_file('config
/*.yml', true) | lua vim.api.nvim_get_runtime_file('config/*.json', true) " "+:q"
```
take

### Expected behavior

Both should perform similarly fast.

### Actual behavior

```console
❯ time nvim -u NONE --cmd "lua vim.api.nvim_get_runtime_file('config/*.yaml', true) | lua vim.api.nvim_get_runtime_file('config
/*.yml', true) | lua vim.api.nvim_get_runtime_file('config/*.json', true) " "+:q"

________________________________________________________
Executed in   20.79 millis    fish           external
   usr time    7.76 millis  128.00 micros    7.63 millis
   sys time    3.31 millis   45.00 micros    3.27 millis

~
❯ time nvim -u NONE --cmd "lua vim.api.nvim_get_runtime_file('config/*.{yaml,yml,json}', true)" "+:q"

________________________________________________________
Executed in  148.62 millis    fish           external
   usr time  108.54 millis  158.00 micros  108.38 millis
   sys time   42.69 millis   62.00 micros   42.63 millis
```

With the braces it takes a lot longer, even though it should be doing the same amount of work roughly.

And the difference is a lot bigger when measuring the functions directly:

I made a small test plugin, and it shows a difference of about a factor of 150 (times in µs): 
![grafik](https://user-images.githubusercontent.com/11978847/171619259-d6026ce9-3d5e-4ad7-a7b5-de1a20075c60.png)

The plugin is writen in rust with my own bindings, but someone knowledgeable in Lua should be able to write an easier to run test there as well:
https://github.com/ModProg/nvim-perf-runtime_file/blob/c84070386220cfc43ce300acdd66fa0b3f05690f/src/lib.rs#L12-L28

**Comment by lewis6991** [comment]
My guess is due to the syntax used, it is shelling out to bash to perform the glob. You're probably better off just calling `nvim_get_runtime` several times.

**Comment by zeertzjq** [comment]
Fixed by #24296
